lib/Doctrine/Common/EventManager.php

```diff
@@ -1,45 +1,43 @@
 <?php
 namespace Doctrine\Common;
 
+use function spl_object_hash;
+
 /**
  * The EventManager is the central point of Doctrine's event listener system.
  * Listeners are registered on the manager and events are dispatched through the
  * manager.
- *
- * @link   www.doctrine-project.org
- * @since  2.0
- * @author Guilherme Blanco <guilhermeblanco@hotmail.com>
- * @author Jonathan Wage <jonwage@gmail.com>
- * @author Roman Borschel <roman@code-factory.org>
  */
 class EventManager
 {
     /**
      * Map of registered listeners.
      * <event> => <listeners>
      *
-     * @var array
+     * @var object[][]
      */
     private $_listeners = [];
 
     /**
      * Dispatches an event to all registered listeners.
      *
-     * @param string    $eventName      The name of the event to dispatch. The name of the event is
+     * @param string         $eventName The name of the event to dispatch. The name of the event is
      *                                  the name of the method that is invoked on listeners.
      * @param EventArgs|null $eventArgs The event arguments to pass to the event handlers/listeners.
      *                                  If not supplied, the single empty EventArgs instance is used.
      *
      * @return void
      */
-    public function dispatchEvent($eventName, EventArgs $eventArgs = null)
+    public function dispatchEvent($eventName, ?EventArgs $eventArgs = null)
     {
-        if (isset($this->_listeners[$eventName])) {
-            $eventArgs = $eventArgs === null ? EventArgs::getEmptyInstance() : $eventArgs;
+        if (! isset($this->_listeners[$eventName])) {
+            return;
```

Comment:
Any proposal how to cover this line by a test that would actually assert something?

Reply:
Not sure if it'll count, but have a listener for `foo` that expects to be not called and dispatch `bar` event?

Reply:
That would indeed cover the line (also not registering anything would work), but sounds like cheating. :trollface: 
I got only one dirty idea: Use reflection to assert that EventArgs::$_emptyEventArgsInstance is `null` after dispatching an unknown event.